### ISSUE TITLE
Trying to process more than one image: crypto HashUpdate fail

### ISSUE THREAD

**Issue by EdJr** [martinheidegger/mapslice#10]
I'm trying to implement a page with express and multer where uploaded images should be immediately processed by mapslice. It works fine with the first image, but fails for the second one, whether I try to upload both at the same time or sequentially:

    crypto.js:70
      this._handle.update(data, encoding);
                   ^
    TypeError: HashUpdate fail
    [....]
In nodejs/node-v0.x-archive#1415 it is stated that "crypto instances are not reusable", so I guess the script is trying to reuse the same object for the second image. Is this a bug in mapslice or am I doing something wrong?

The following code reproduces the problem, although it's not exactly what I'm using:

    var mapslice = require('mapslice');

    var mapSlicer = mapslice({
      file: 'myimg.png'
    })
      .on("start", function(files, options) {
        console.info("Processing first image....");
    })
      .on("end", function() {
        console.info("Finished processing first image.\n\nProcessing second image....");

        var mapSlicer2 = mapslice({
          file: 'myimg.png'
        }).start();
    }).start();

**Comment by EdJr** [comment]
Alright, I just looked at MapSlicer.js. I removed line 15 (`sha1 = crypto.createHash('sha1');`) and changed line 122 to (`patternMd5 = crypto.createHash('sha1').update(this.options.output).digest("hex");`) and now it seems to be working. Is this a right change for a pull?

**Comment by martinheidegger** [comment]
Thank you for fixing it. It is a slightly weird way to create PR though the code landed in master and is published as 1.0.3 @ npm. Again thanks and best regards!

**Comment by EdJr** [comment]
No problem! It makes me happy to know I could finally help with something, even it being so simple, for the first time in my life.

Sorry for the noobish questions, but I don't understand what you mean by "It is a slightly weird way to create PR". What is PR? Are you referring to the way it was before of after I changed the code?

Best regards!

**Comment by martinheidegger** [comment]
A "PR" or "Pull Request" is the-github-way™ to collaborate. Github has a guide that explains it [here](https://help.github.com/articles/using-pull-requests/#initiating-the-pull-request)

**Comment by EdJr** [comment]
Oh, of course. I thought it had something to do with mapslice itself. I should have thought a little more before asking.

Now, I promise this is the last question, but I really want to learn the right way to do this stuff. What was weird about the way I created the PR? I've already the guide before, but I'm not sure. Is it because I opened an issue and linked it to the PR? If so, I apologize. I only thought about trying to fix the bug myself after opening the issue.
